src/components/valences-test/ValencesTest.tsx

```diff
@@ -16,202 +15,190 @@ import TestResults from "../test-results/TestResults";
 import { getValencesTestSettings } from "./settings/ValencesTestSettings";
 import "./ValencesTest.scss";
 
-type Props = RouteComponentProps<any> & React.Props<any>;
-
-interface IValencesTestQuestionCard extends Question {
+interface ValencesTestQuestionCard extends Question {
   data: IElement;
 }
 
-interface IValencesTestState {
-  questions: IValencesTestQuestionCard[];
-  right: IValencesTestQuestionCard[];
-  wrong: IValencesTestQuestionCard[];
+interface ValencesTestState {
+  questions: ValencesTestQuestionCard[];
+  right: ValencesTestQuestionCard[];
+  wrong: ValencesTestQuestionCard[];
 }
 
-@autobind
-class ValencesTest extends React.Component<Props> {
-  public state: IValencesTestState = {
-    questions: [],
-    right: [],
-    wrong: [],
-  };
-
-  private settings: IValencesTestSettings = getValencesTestSettings();
+function ValencesTest() {
+  const settings = React.useRef(getValencesTestSettings() || []).current;
```

Comment:
Is the intention here to make `settings` a singleton for the life of the component? A `useState` with an initial state function and no setter is a marginally better pattern for that as it won't call the function every render:

```ts
const [settings] = React.useState(() => getValencesTestSettings() || []);
```

but perhaps `useMemo` works the best in this case?

```ts
const settings = React.useMemo(() => getValencesTestSettings() || [], []);
```

Reply:
Just looking at the `getValencesTestSettings`, I don't think the `|| []` is needed here at all.

Reply:
Oh yeah that was part of a test i was doing, will remove

Reply:
Yeah useMemo can work here too, less reliable then useRef and useState, because React won't guarantee its referential stable always, there was no performance issue with calling getValneceTest on the useRef initializer, that why I didn't optimize to useState

Reply:
I think we can just move to useMemo, if needs momization at all, maybe ill just move outside the component